lib/mongo.ex

```diff
@@ -185,6 +185,21 @@ defmodule Mongo do
   end
 
   @doc """
+  Selects a single document in a collection and return the document or null
+  """
+  @spec findOne(Pool.t, collection, BSON.document, Keyword.t) :: BSON.document
+  def findOne(pool, coll, filter, opts \\ []) do
+    opts = Dict.put(opts, :limit, 1)
+    cursor = find(pool, coll, filter, opts)
```

Comment:
We shouldn't call `find` because we don't want the extra overhead of using a cursor. It should be implemented directly in this function.